proposals/PluginsAudit2015/audit.md

```diff
@@ -0,0 +1,223 @@
+* rename to cordova-plugin-device-orientation-motion?
+* low priority: polish code, update code to remove accelerometer references. 
+* Recommendation: Keep. Code needs a cleanup but low priority.
+
+## Dialogs
+* 1 Month Downloads: 12456
+* Open Issues: 13
+* Open Pull Requests: 17
+* Using custom styled divs should be used instead of alerts
+* In HTML 5.1 there is a proposed dialog element which actually seems pretty good. http://www.w3.org/html/wg/drafts/html/master/single-page.html#the-dialog-element 
+* looks good on android
+* looks ugly on ios
+* currently not sync which is how default alerts work 
+* window.alert still available but says index.html
+* Recommendation: Deprecate! Encourage use of HTML/CSS/JS divs instead.
+
+## File
+* 1 Month Downloads: 21252
+* Open Issues: 60
+* Open Pull Requests: 15
+* phonegap-plugin-file or phonegap-plugin-fs. Consumes cordova-plugin-file. Create downstream version with a better interface. Current one is based on standard which is ugly. (rip shelljs file ops)
+* Can we treat some urls consistency across all platforms? Possibly use Flex Air package for inspiration.
+* ex: app:, temp:, 
+* cdvfile wasn’t consistent enough across platforms
+* Recommendation: Keep. Work on creating more consistency across platforms. Especially with cdvfile. Create phonegap-plugin-file with improved api.
+
+## File Transfer
+* 1 Month Downloads: 16597
+* Open Issues: 72
+* Open Pull Requests: 8
+* xhr2 is an alternative
+* available on all the platforms now. 
+* http://caniuse.com/#search=xhr2
+* Recommendation: Keep for now, but also start recommending xhr2 instead. Create tutorials using xhr2 instead of file-transfer.
+
+## Geolocation
+* 1 Month Downloads: 14709
+* Open Issues: 35
+* Open Pull Requests: 10
+* Spec: http://dev.w3.org/geo/api/spec-source.html 
+* Recommendation: Keep.
+
+## Globalization
+* 1 Month Downloads: 5108
+* Open Issues: 12
+* Open Pull Requests: 0
+* Deprecate async calls
+* query all these values at startup and provide a synchronous API to get these values
+* Maybe it would be better to have JS libraries handle this?
+* Recommendation: Keep. Deprecate async calls and make plugin sync.
+
+## In-App-Browser
+* 1 Month Downloads: 17904
+* Open Issues: 116
+* Open Pull Requests: 36
+* API needs a refactor
+* Cordova Webview based inappbrowser as a replacement would have big security issues if we opened up third party links in it. So we shouldn’t do it.
+* Carlos suggested new inappbrowser based on safariview. He can elaborate more.
+* encourage people to use it through apis instead of window.open shims
+* demo of oauth workflow + demo of closing inappbrowser via js (url change)
+* http://ngcordova.com/docs/plugins/oauth/
+* get rid of insertcss & executescript. Use fileapi instead
+* Recommendation: Keep. Needs a refactor. Remove insertcss & executescript. Encourage use through apis instead of window.open. Make demos doing oAuth. Maybe phonegap-plugin-oauth.
```

Comment:
Something based on SFSafariViewController (iOS) and Chrome Custom Tabs (Android) would be ideal. Some OAuth providers (Fitbit, for instance) have started adding restrictions that you're not allowed to do the OAuth flow in non-browser web views.